tinkerpop3/src/main/java/io/shiftleft/overflowdb/OdbGraphBuilder.java

```diff
@@ -0,0 +1,87 @@
+package io.shiftleft.overflowdb;
+
+import gnu.trove.map.TLongObjectMap;
+import gnu.trove.map.hash.TLongObjectHashMap;
+import org.apache.tinkerpop.gremlin.structure.VertexProperty;
+
+import java.util.ArrayList;
+
+public class OdbGraphBuilder {
+  private final OdbGraph graph;
+  TLongObjectMap<NodeRef> newNodes = new TLongObjectHashMap<>();
+  TLongObjectHashMap<ArrayList<EdgeInfo>> edges = new TLongObjectHashMap<>();
+  TLongObjectMap<ArrayList<PropertyInfo>> nodeProperties = new TLongObjectHashMap<>();
```

Comment:
what about `edgeProperties`?

Reply:
we don't use them in CpgLoader, they are loaded when the edge is created.

Reply:
this is a standalone generic database though, why not add it while we're on it?